policyengine_api/endpoints/policy.py

```diff
@@ -198,45 +206,66 @@ def get_policy_search(country_id: str) -> list:
     if country_not_found:
         return country_not_found
 
-    results = database.query(
-        "SELECT id, label, policy_hash FROM policy WHERE country_id = ? AND label LIKE ?",
-        (country_id, f"%{query}%"),
-    )
-    if results is None:
-        return dict(
-            status="error",
-            message=f"Policy not found in {country_id}",
+    try:
+        results = database.query(
+            "SELECT id, label, policy_hash FROM policy WHERE country_id = ? AND label LIKE ?",
+            (country_id, f"%{query}%"),
         )
-    else:
+        if results is None:
+            body = dict(
+                status="error",
+                message=f"Database query failed for {country_id} with query '{query}",
+            )
+            return Response(
+                json.dumps(body), status=500, mimetype="application/json"
+            )
+
         results = results.fetchall()
 
-    # If unique_only is true, filter results to only include
-    # items where everything except ID is unique
-    if unique_only:
-        processed_vals = set()
-        new_results = []
-
-        # Compare every label and hash to what's contained in processed_vals
-        # If a label-hash set aren't already in processed_vals,
-        # add them to new_results
-        for policy in results[:]:
-            comparison_vals = policy["label"], policy["policy_hash"]
-            if comparison_vals not in processed_vals:
-                new_results.append(policy)
-                processed_vals.add(comparison_vals)
-
-        # Overwrite results with new_results
-        results = new_results
-
-    # Format into: [{ id: 1, label: "My policy" }, ...]
-    policies = [
-        dict(id=result["id"], label=result["label"]) for result in results
-    ]
-    return dict(
-        status="ok",
-        message=None,
-        result=policies,
-    )
+        if not results:
```

Comment:
This if-block is functionally the same as the above one.

`if not results` is broader in Python than `if results is None`, but they're very similar, and in this case, basically the same. I would remove the if-block beginning with `if not results:`

Reply:
`results is None` will not catch an empty  list, so the problem you raised in #416 remains, i.e, if I query, say :
`api.policyengine.org/ng/policies?query=tax` it will return:

```{
    "status": "ok",
    "message": "Policies found",
    "result": []
}```